hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1328,55 +1275,145 @@ void lower(Vunit& unit, cvttsd2siq& i, Vlabel b, size_t idx) {
   });
 }
 
+void lower(Vunit& u, callm& i, Vlabel b, size_t z) {
+  lower_impl(u, b, z, [&] (Vout& v) {
+  lowerVptr(i.target, v);
+
+  auto const scratch = v.makeReg();
+
+  // Load the target from memory and then call it
+  v << load{i.target, scratch};
+  v << callr{scratch, i.args};
+  });
+}
+
+void lower(Vunit& u, jmpm& i, Vlabel b, size_t z) {
+  lower_impl(u, b, z, [&] (Vout& v) {
+  lowerVptr(i.target, v);
+
+  auto scratch = v.makeReg();
+
+  v << load{i.target, scratch};
+  v << jmpr{scratch, i.args};
+  });
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+void lower(Vunit& u, callstub& i, Vlabel b, size_t z) {
+  lower_impl(u, b, z, [&] (Vout& v) {
+    v << call{i.target, i.args};
+  });
+}
+
+void lower(Vunit& u, callfaststub& i, Vlabel b, size_t z) {
```

Comment:
(I guess the heart of the issue is that if we ever changed the implicit defs of `callstub{}` and `callfaststub{}` to be anything but a subset of `call{}`, this would be a bug.  I think `callm{}` and `call{}` can be assumed to have the same implicit defs, though, so that one is fine.)